src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Helpers/DelegationHelper.cs

```diff
@@ -275,5 +275,45 @@ public static string GetAttributeMatchKey(List<AttributeMatch> attributeMatches)
         {
             return string.Concat(attributeMatches.OrderBy(r => r.Id).Select(r => r.Id + r.Value));
         }
+
+        /// <summary>
+        /// Sets the RuleType on each rule in the given list
+        /// </summary>
+        public static void SetRuleType(List<Rule> rulesList, List<int> keyRolePartyIds, List<AttributeMatch> coveredBy, int parentPartyId = 0)
+        {
+            foreach (Rule rule in rulesList)
+            {
+                if (TryGetDelegationParamsFromRule(rule, out _, out _, out _, out int? coveredByPartyId, out int? coveredByUserId, out _)
+                    && rule.Type == RuleType.None)
+                {
+                    SetTypeForSingleRule(keyRolePartyIds, coveredBy, parentPartyId, rule, coveredByPartyId, coveredByUserId);
+                }
+            }
+        }
+
+        private static void SetTypeForSingleRule(List<int> keyRolePartyIds, List<AttributeMatch> coveredBy, int parentPartyId, Rule rule, int? coveredByPartyId, int? coveredByUserId)
+        {
+            if ((TryGetCoveredByUserIdFromMatch(coveredBy, out int coveredByUserIdFromRequest) && coveredByUserIdFromRequest == coveredByUserId)
+                || (TryGetCoveredByPartyIdFromMatch(coveredBy, out int coveredByPartyIdFromRequest) && coveredByPartyIdFromRequest == coveredByPartyId))
+            {
+                rule.Type = RuleType.DirectlyDelegated;
+            }
+            else if (TryGetCoveredByUserIdFromMatch(coveredBy, out _) && keyRolePartyIds.Any(id => id == coveredByPartyId))
+            {
+                rule.Type = RuleType.InheritedViaKeyRole;
+            }
+            else if (parentPartyId == coveredByPartyId)
```

Comment:
Should perhaps the different values that are compared in this method also be checked for not being 0?
Could we not otherwise perhaps get false type results where 0=0?

Reply:
Also I think the check might be wrong. Type.InheritedAsSubunit is when the OfferedBy of the rule is the parentPartyId, and the coveredBy is the direct CoveredBy Userid or PartyId of the request.

Reply:
I fixed the check so the type is set to InheritedAsSubunit only if the rule is directly delegated and the Offeredy and parentPartyId are the same.